test/integration/connection-monitoring-and-pooling/connection_monitoring_and_pooling.test.ts

```diff
@@ -16,4 +20,37 @@ describe('Connection Monitoring and Pooling (Node Driver)', function () {
       }
     ]
   });
+
+  describe('ConnectionPoolCreatedEvent', () => {
+    let client: MongoClient;
+    beforeEach(async function () {
+      client = this.configuration.newClient();
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    describe('constructor()', () => {
+      it('when auth is enabled redacts credentials from options', {
```

Comment:
Should we add a test to confirm we're not modifying the `options` on the connection pool?

Reply:
Added a check